src/main/kotlin/app/hashers/CommitObservable.kt

```diff
@@ -0,0 +1,114 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Anatoly Kislov (anatoly@sourcerer.io)
+
+package app.hashers
+
+import app.Logger
+import app.model.Commit
+import app.model.DiffContent
+import app.model.DiffFile
+import app.model.DiffRange
+import app.model.Repo
+import app.utils.RepoHelper
+import io.reactivex.Observable
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.RawText
+import org.eclipse.jgit.errors.MissingObjectException
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.util.io.DisabledOutputStream
+import java.nio.charset.Charset
+
+object CommitObservable {
+    fun getObservable(git: Git, repo: Repo) = Observable
+        .create<Commit> { subscriber ->
+            try {
+                val revWalk = RevWalk(git.repository)
+                val commitId = git.repository.resolve(RepoHelper.MASTER_BRANCH)
+                revWalk.markStart(revWalk.parseCommit(commitId))
+                for (revCommit in revWalk) {
+                    subscriber.onNext(Commit(revCommit))
+                }
+                // Commits are combined in pairs, an empty commit concatenated
+                // to calculate the diff of the initial commit.
+                subscriber.onNext(Commit())
+            } catch (e: Exception) {
+                Logger.error("Commit producing error", e)
+                subscriber.onError(e)
+            }
+            subscriber.onComplete()
+        }  // TODO(anatoly): Rewrite diff calculation in non-weird way.
```

Comment:
It's a funny TODO item but not clear what exactly you want to do here. It's up to you to replace on something more descriptive or just ignore it.